### ISSUE TITLE
Parallelization

### ISSUE THREAD

**Issue by aaronrudkin** [DeclareDesign/fabricatr#26]
We want to add parallelization to resample and fabricate. This will be a dummy issue to track progress.

- [ ] Allow users to provide a parallelization backend
- [ ] Allow users to ask use to provide a parallelization backend
- [ ] Ensure sane seed / RNG behavior across parallelization
- [ ] Benchmark parallelization to see under what cases this actually provides reasonable additional speed
- [ ] Figure out testing options to reflect parallelization across operating systems

**Comment by graemeblair** [comment]
Thanks, Aaron. I get the parallelization in resample, what's the parallelization in fabricate?

**Comment by aaronrudkin** [comment]
@graemeblair Now that we (will) allow non-nested data, the mutually exclusive data frames may gain some benefit from parallelization. (i.e. in Neal's story, we can generate the PS+PC and SS+SC silos in parallel)

I would suspect the degree of parallelization would generally lower and this is certainly a lower importance case than resampling.

**Comment by graemeblair** [comment]
sounds good; yes seems like a lower priority but worth exploring too

**Comment by graemeblair** [comment]
Given how often levels refer to each other (including just by N) and Aaron's edit knowing this will make speed gains unlikely, closing this for now
